lib/Receiver.js

```diff
@@ -352,20 +354,28 @@ class Receiver {
    */
   dataMessage () {
     if (this.fin) {
-      const buf = this.fragments.length > 1
-        ? Buffer.concat(this.fragments, this.messageLength)
-        : this.fragments.length === 1
-          ? this.fragments[0]
-          : constants.EMPTY_BUFFER;
-
+      const fragments = this.fragments;
+      const messageLength = this.messageLength;
       this.totalPayloadLength = 0;
-      this.fragments.length = 0;
+      this.fragments = [];
       this.messageLength = 0;
       this.fragmented = 0;
 
       if (this.opcode === 2) {
-        this.onmessage(buf, { masked: this.masked, binary: true });
+        let data;
```

Comment:
I know for sure that [compund let assignment](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#2-unsupported-syntax) are not optimizable on older V8 versions.

I also saw deoptimizations for non compound assignment when rewriting the parser so for now I prefer to stick with `var` when in doubt. This same is also done in Node.js core.